.eslintrc.js

```diff
@@ -176,9 +185,22 @@ module.exports = {
 				'@typescript-eslint/restrict-template-expressions': 'off',
 				'@typescript-eslint/unbound-method': 'off',
 
+				// override @typescript-eslint/stylistic-type-checked to ignore consistent-indexed-object-style. Both are used as required.
+				'@typescript-eslint/consistent-indexed-object-style': 'off',
+
 				// force explicit member accessibility modifiers
 				'@typescript-eslint/explicit-member-accessibility': 'error',
 
+				// override @typescript-eslint/stylistic-type-checked to ignore booleans in nullish coalescing checks
+				// https://typescript-eslint.io/rules/prefer-nullish-coalescing#ignoreprimitives
+				'@typescript-eslint/prefer-nullish-coalescing': [
+					'error',
+					{ ignorePrimitives: { boolean: true } },
+				],
+
+				// ban non-null assertions
+				'@typescript-eslint/no-non-null-assertion': 'error',
```

Comment:
I usually try to group related rules or keep them at least alphabetically ordered. So I would flip-flop these two for readability.